### DISCUSSION TITLE
✅ Unable to parse private key

### DISCUSSION THREAD

**Discussion by TheCableGuy99** [isontheline/pro.webssh.net#915] [General]
Hi,

I'm trying to trial this on my iPad and I've added my private SSH Key as follows (edited so not my actual key):

PuTTY-User-Key-File-2: ecdsa-sha2-nistp384
Encryption: aes256-cbc
Comment: blah.blah@hotmail.co.uk
Public-Lines: 3
AAAAE2VjZHNhLXNoYHJ£N£JIhnjncrnrjcmnvIbmlzdHAzODQAAABhBIQc/dsjn*YB
mkfh7vSTqOMFdcnwejn4UUNuh4b4ufn/Tv6oPg9LwfTqTBsr9mK/zG2vBFky
vU96fwFK1Qxsa7jDadpPYFfskawMd2rGHjUoysRpdYDbwH/jac/vdw==
Private-Lines: 2
OIH*X£Nuhce0hceiehhceor+GAe1SLHNewkjnc3UBIUBIUE/wQHhgYdd+qLW4eJgn
I8ewn3b98YH98hncxn39h33hnc3==
Private-MAC: b3ac3fcbd1843hnfd39hd3hdh39fba46c96d4e4e8e64

When trying to connect I get the error that it was "Unable to Parse the Private Key". I've entered my private key passphrase and it's fine and the information in the app says it supports these types of keys (Putty ECDSA) so I'm not sure where I'm going wrong.

Can anyone assist please?

**Comment by isontheline** [comment]
Hi @TheCableGuy99 👋

Thank you a lot for your feedback 🙏

Please enable verbose logging and retry.
Perhaps we will find some logs of the root cause.

In order to enable verbose logging you can follow the steps :
1. iOS Settings
2. WebSSH
3. Settings
4. Log Level > Verbose
5. Restart WebSSH
6. Reproduce your action
7. Find the log files inside iOS > Files > WebSSH

Have a nice day ☀️

**Reply by TheCableGuy99** [reply]
Hey,

Thanks for replying.

I've done as you suggested but there is no WebSSH folder in the Files app.

**Reply by TheCableGuy99** [reply]
Found it using a different file manager

[log-F13334.txt](https://github.com/isontheline/pro.webssh.net/files/11715505/log-F13334.txt)

**Reply by isontheline** [reply]
This seems to be a pkey format error :

> <error>No Private-MAC</error>

Private MAC seems not readable by WebSSH but I don't know why.

Are you able to upgrade your PuTTY key to the format number "3" three?

**Reply by TheCableGuy99** [reply]
Hey, 

Well I wasn't aware there were version numbers for keys, do you have any kind of example what you are asking me to try?

Thanks.

**Reply by isontheline** [reply]
Perhaps you should try this to convert your private key to PEM :
https://stackoverflow.com/a/2224204

**Comment by TheCableGuy99** [comment]
Okay, I just created a new Ed25519 key instead and it's worked this time. Maybe my key was just a little old?

Bonus: Even works with my Yubikey :)

Thanks for your help!

**Comment by isontheline** [comment]
You’re welcome !

Thank you so much for your feedback 🙏
